frontend/src/metabase/visualizations/components/settings/ChartSettingsTableFormatting.jsx

```diff
@@ -426,9 +429,9 @@ const RuleEditor = ({
               ...COMMON_OPERATOR_NAMES,
               ...(isBooleanRule
                 ? BOOLEAN_OPERATIOR_NAMES
-                : isNumericRule
+                : !isKeyRule && isNumericRule
```

Comment:
Wdyt about migrating nested ternaries to `ts-pattern` for better readability?

Reply:
I don't think it'll make this clearer, since ts-pattern is primarily made to work with a limited number of objects taking various shapes, and we're dealing with several variables here. I can shoehorn this into ts-pattern but I wouldn't count on it to improve readability.